src/figuresClasses.ts

```diff
@@ -1,19 +1,94 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape : Shape,
+  color : Color,
+  getArea() :number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape:Shape = 'triangle';
 
-}
+  constructor(
+    public color:Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('any triangle`s side should be larger than zero');
+    }
 
-export class Circle {
+    const longestSide = Math.max(a, b, c);
 
+    if (longestSide >= (a + b + c) - longestSide) {
+      throw new Error('longest side should be larger than'
+      + 'sum of the other sides');
+    }
+  }
+
+  getArea(): number {
+    const halfPerimeter = (this.a + this.b + this.c) / 2;
+    const square = Math.sqrt(halfPerimeter * (halfPerimeter - this.a)
+     * (halfPerimeter - this.b)
+     * (halfPerimeter - this.c));
+
+    return Math.round(square * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape:Shape = 'circle';
+
+  constructor(
+    public color:Color,
+    public radius: number,
+  ) {
+    this.color = color;
+    this.radius = radius;
```

Comment:
if you use public with arg it will be the same property in costructor. so you can delete  this.color = color;
    this.radius = radius;